pkg/api/types/common.go

```diff
@@ -0,0 +1,33 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package types
+
+import (
+	"context"
+	"io"
+)
+
+type CommonOptions struct {
+	// Context is the cobra.Context
+	Context context.Context
```

Comment:
I don't think this is a good pattern since "storing context in struct" is discouraged:

https://go.dev/blog/context-and-structs

https://pkg.go.dev/context#:~:text=Do%20not%20store%20Contexts%20inside%20a%20struct%20type%3B%20instead%2C%20pass%20a%20Context%20explicitly%20to%20each%20function%20that%20needs%20it.%20The%20Context%20should%20be%20the%20first%20parameter%2C%20typically%20named%20ctx%3A

Reply:
and the `GlobalCommandOptions` and `CommonOptions` are confusing (at first they look similar).

I would suggest removing this `CommonOptions` also because they are not "options" (instead, context, io.writer, etc)

Reply:
SGTM